app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupUnlockNavigation.kt

```diff
@@ -3,27 +3,71 @@ package com.x8bit.bitwarden.ui.auth.feature.accountsetup
 import androidx.navigation.NavController
 import androidx.navigation.NavGraphBuilder
 import androidx.navigation.NavOptions
+import androidx.navigation.NavType
+import androidx.navigation.navArgument
 import com.x8bit.bitwarden.ui.platform.base.util.composableWithPushTransitions
 
 /**
- * Route for [SetupUnlockScreen]
+ * Route constants for [SetupUnlockScreen]
  */
-const val SETUP_UNLOCK_ROUTE = "setup_unlock"
+private const val SETUP_UNLOCK_PREFIX = "setup_unlock"
+private const val SETUP_UNLOCK_AS_ROOT_PREFIX = "${SETUP_UNLOCK_PREFIX}_as_root"
+const val SETUP_UNLOCK_INITIAL_SETUP_ARG = "isInitialSetup"
+const val SETUP_UNLOCK_AS_ROOT_ROUTE = "$SETUP_UNLOCK_AS_ROOT_PREFIX/" +
+    "{$SETUP_UNLOCK_INITIAL_SETUP_ARG}"
+private const val SETUP_UNLOCK_ROUTE = "$SETUP_UNLOCK_PREFIX/{$SETUP_UNLOCK_INITIAL_SETUP_ARG}"
 
 /**
  * Navigate to the setup unlock screen.
  */
 fun NavController.navigateToSetupUnlockScreen(navOptions: NavOptions? = null) {
-    this.navigate(SETUP_UNLOCK_ROUTE, navOptions)
+    this.navigate("$SETUP_UNLOCK_PREFIX/false", navOptions)
 }
 
 /**
- * Add the setup unlock screen to the nav graph.
+ * Navigate to the setup unlock screen as root.
  */
-fun NavGraphBuilder.setupUnlockDestination() {
+fun NavController.navigateToSetupUnlockScreenAsRoot(navOptions: NavOptions? = null) {
+    this.navigate("$SETUP_UNLOCK_AS_ROOT_PREFIX/true", navOptions)
+}
+
+/**
+ * Add the setup unlock screen to a nav graph.
+ */
+fun NavGraphBuilder.setupUnlockDestination(
+    onNavigateBack: () -> Unit,
+) {
     composableWithPushTransitions(
```

Comment:
Should this one be a slide up?

Reply:
I am not sure, but looks like other screens in this graph use that so I will say "sure, I'm Ron Burgandy?"

Reply:
I'm not sure on this one, maybe a quick question over at design.